src/components/ProfileList/ModerationProfileListItem.tsx

```diff
@@ -13,19 +11,53 @@ import { JobSpecialization } from '@/components/ProfileList/profile-data'
 import TechnologiesRenderer from '@/components/renderers/TechnologiesRenderer'
 import AcceptIcon from '@/assets/icons/AcceptIcon'
 import RejectIcon from '@/assets/icons/RejectIcon'
+import { ToastStatus, useToast } from '@/contexts/ToastContext'
+import { apiClient } from '@/lib/apiClient'
+import { useAsyncAction } from '@/hooks/useAsyncAction'
+
+import classNames from 'classnames/bind'
+import styles from '@/components/ProfileList/ProfileList.module.scss'
+import { useModal } from '@/contexts/ModalContext'
 
 const cx = classNames.bind(styles)
 
-const stateStatus = (state: string): React.ReactNode => {
+type StateStatusProps = {
+  profile: ProfileModel
+}
+
+function StateStatus({ profile }: StateStatusProps) {
+  const { id, state } = profile
+  const { addToast } = useToast()
+  const { setProfileId, setShowRejectModal } = useModal()
```

Comment:
**nitpick:** Instead of keeping profile id in modal provider, just render this Reject Modal here, and pass `profileId` as prop.